prov/rxm/src/rxm_eq.c

```diff
@@ -64,6 +64,74 @@ static struct fi_ops rxm_eq_fi_ops = {
 	.ops_open = fi_no_ops_open,
 };
 
+ssize_t rxm_eq_write(struct fid_eq *eq_fid, uint32_t event,
+		     const void *buf, size_t len, uint64_t flags)
+{
+	struct fi_eq_entry entry;
+	const struct fi_eq_entry *in_entry = buf;
+	struct rxm_mc *rxm_mc;
+	int ret;
+
+	if ((event != FI_JOIN_COMPLETE) && (event != FI_JOIN_FAILED)){
+		return ofi_eq_write(eq_fid, event, buf, len, flags);
+	}
+
+	rxm_mc = in_entry->context;
+
+	memset(&entry, 0, sizeof(entry));
+	entry.context = rxm_mc->mc_fid.fid.context;
+	entry.fid = &rxm_mc->mc_fid.fid;
+
+	if (event == FI_JOIN_FAILED)
+		goto error;
+
+	ofi_spin_lock(&rxm_mc->state_lock);
+	if (rxm_mc->state == RXM_MC_UTIL_STARTED) {
+		if (!rxm_mc->ep->offload_coll_ep) {
+			rxm_mc->state = RXM_MC_READY;
+			ofi_spin_unlock(&rxm_mc->state_lock);
```

Comment:
missing lock -- do not acquire and release locks in separate function calls.  That is very difficult to maintain and check for correct behavior.

Reply:
Done.